astropy/coordinates/representation/base.py

```diff
@@ -426,6 +451,37 @@ def shape(self, shape):
                 else:
                     reshaped.append(val)
 
+    @property
+    def masked(self):
+        return isinstance(getattr(self, self.components[0]), Masked)
```

Comment:
I know that it's designed s.t. f one field is masked all are masked, but with early exiting the following should be equivalent speed-wise 🤞🏼

```suggestion
        return any(isinstance(getattr(self, c), Masked) for c in self.components)
```

Reply:
Not sure: if this gets out of sync, there are arguably larger problems than whether `.masked` is still OK - e.g., any assignments with masked values will silently drop the mask (numpy problem, not something easy to catch...).

But happy to change it too...

Reply:
It's not a large priority, we're just not built around immutable data structures.